simulator/include/cache.h

```diff
@@ -0,0 +1,104 @@
+#ifndef CHS_CACHE_H
+#define CHS_CACHE_H
+
+#include <cinttypes>
+#include <vector>
+#include <memory>
+
+#include "types.h"
+
+namespace CacheHierarchySimulator
+{
+
+struct CacheEntry
+{
+    bool valid = 0;
+    uint32_t tag = 0;
+};
+
+enum ReplacementPolicy
+{
+    FIFO,
+    LIFO,
+    LRU,
+    PLRU,
+    MRU,
+    LFU
+};
+
+// Typedefs
```

Comment:
Would it be better if all these typesdefs were in "types.h"

Reply:
Possibly. My thought was that since they are only relevant in the Cache class, that they remain there. I can see it the other way, though.

Reply:
I'm good either way. No need to update anything if these are only used in the Cache class.